avogadro/core/connectedgroup.cpp

```diff
@@ -0,0 +1,161 @@
+/******************************************************************************
+  This source file is part of the Avogadro project.
+  This source code is released under the 3-Clause BSD License, (see "LICENSE").
+******************************************************************************/
+
+#include "connectedgroup.h"
```

Comment:
I guess my main question is philosophical. Do you need this as a separate class rather than adding / modifying the graph class? Some of this seems like it duplicates work that the graph class already does.

Reply:
I will use this class to control the groups in the layering system. Maybe it can be inside the graph file. But this has more use outside the graph idea.